graylog2-server/src/test/java/org/graylog2/shared/buffers/processors/MessageTimestampTest.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+package org.graylog2.shared.buffers.processors;
+
+import org.graylog2.plugin.Message;
+import org.graylog2.plugin.MessageFactory;
+import org.graylog2.plugin.TestMessageFactory;
+import org.graylog2.plugin.Tools;
+import org.joda.time.DateTime;
+import org.junit.jupiter.api.Test;
+
+import java.time.Duration;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+class MessageTimestampTest {
```

Comment:
These tests don't test the changed behavior of the ProcessBufferProcessor, I'd greatly prefer if we tested the overall change to the codebase as well as just the added method.

Reply:
Revised to call through the `ProcessBufferProcessor` handler code path.